### ISSUE TITLE
FurnitureWorkshopTest writes to disk

### ISSUE THREAD

**Issue by mikejbrown** [TeamPorcupine/ProjectPorcupine#1344]
These lines in [FurnitureWorkshopTest](https://github.com/TeamPorcupine/ProjectPorcupine/blob/master/Assets/Editor/UnitTests/Models/FurnitureWorkshopTest.cs#L61) are bad juju:

```c#
FileStream fs = new FileStream("Workshop.xml", FileMode.Create);
TextWriter writer = new StreamWriter(fs, new UTF8Encoding());

XmlSerializer serializer = new XmlSerializer(typeof(FurnitureWorkshop));
serializer.Serialize(writer, fi);
writer.Close();

FileStream fr = new FileStream("Workshop.xml", FileMode.Open);
var dfi = (FurnitureWorkshop)serializer.Deserialize(fr);
```

These lines create a file `Workshop.xml` which is (a) not cleaned up and (b) not necessary to begin with! Unit tests should avoid touching the disk whenever possible since what is being tested here is the xml (de)serialization logic itself and not the ability to create/read files! This is easy enough to fix in this instance using a `StringWriter` instead of files. See, e.g., [SchedulerEditorTest](https://github.com/TeamPorcupine/ProjectPorcupine/blob/master/Assets/Editor/UnitTests/Models/SchedulerEditorTest.cs#L309) for an implementation of this idea.

**Comment by dusho** [comment]
Will fix this when I get to my PC..

**Comment by mikejbrown** [comment]
I'm tempted to add the high priority label simply because the Workshop.xml working tree spam is _really_ annoying me at the moment, but that really probably doesn't justify the label. :grinning:

**Comment by sboigelot** [comment]
@dusho you should just use a MemoryStream

**Comment by mikejbrown** [comment]
Fixed by #1365
